pkg/controllers/manager.go

```diff
@@ -58,14 +57,18 @@ func NewManagerOrDie(config *rest.Config, options controllerruntime.Options) Man
 
 func (m *GenericControllerManager) Register(controllers ...Controller) Manager {
 	for _, controller := range controllers {
-		var builder = controllerruntime.NewControllerManagedBy(m).For(controller.For())
+		controlledObject := controller.For()
+		var builder = controllerruntime.NewControllerManagedBy(m).For(controlledObject)
+		if namedController, ok := controller.(NamedController); ok {
```

Comment:
only lines 62-64 need to change. Unless you're making a wider argument that using controller.For() in multiple places is a bad idea.

Reply:
Yeah. I saw it was being used multiple times. So if the resource isn't changing, I don't see a need to call it multiple times.

Reply:
It's mostly about cleanliness here. This function is only called at boot up, and adds mere microseconds to the process. I'd probably trade one fewer lines of code (and variable names/assignments) for these microseconds. 

If you want to get into the business of shaving off microseconds, we can find much more impactful premature optimizations than this elsewhere in the codebase. This is a deep rabbit hole with no prize at the end, unfortunately. There are definitely times where it's worth sacrificing readability for performance -- not here though.

If you want to make a case that it reduces code readability to use controller.For() multiple times in this function, I'm open this argument.